### ISSUE TITLE
suggestions

### ISSUE THREAD

**Issue by FaridSafi** [jeanpan/react-native-camera-roll-picker#3]
Hello, great component!
I am using it in a project and I suggest the following features:
- loading indicator
- custom circle-check image

I will try to prepare a pull request

**Comment by superandrew213** [comment]
What about [#4 ](https://github.com/jeanpan/react-native-camera-roll-picker/pull/4) for custom selected image marker?

**Comment by jeanpan** [comment]
@superandrew213 Thanks for the PR, will merge it after checking the functionality.

**Comment by jeanpan** [comment]
#4 : Custom circle-check image has been merged.

**Comment by superandrew213** [comment]
Great!

**Comment by superandrew213** [comment]
How about we use ListView to improve performance? Or even better this [react-native-sglistview](https://github.com/sghiassy/react-native-sglistview).

If you scroll past 500 photos memory is hitting +1.1gb

**Comment by stoffern** [comment]
move it to ES6 syntax

**Comment by jeanpan** [comment]
@superandrew213. Ok, I will try to use ListView instead in my free time. Thanks for the suggestion.

**Comment by jiachaosun** [comment]
I try to use ListView,but not easy.I don't know how to recall the renderRow method to show "selectedMarker" component,because when i call setState method the ListView only rerender by render().The renderRow props function not recall.

**Comment by stoffern** [comment]
@jiachaosun I have allready put together a SGListView here https://github.com/jeanpan/react-native-camera-roll-picker/pull/7

if you want to test it right away try this until its merged:
`npm i -S github:stoffern/react-native-roll-picker`

**Comment by jeanpan** [comment]
@FaridSafi : loading indicator has been added in the latest version. please check it !

**Comment by superandrew213** [comment]
Video support would be nice. Any ideas how to go about it? ```CameraRoll``` only returns photos.

**Comment by stoffern** [comment]
@superandrew213 
assetType : The asset type, one of 'Photos', 'Videos' or 'All'. (Default: Photos)

**Comment by superandrew213** [comment]
Lol mixed it up with this [react-native-image-crop-picker](https://github.com/ivpusic/react-native-image-crop-picker)

**Comment by stoffern** [comment]
ah :+1: 
then i would use this picker and my cropper :)
https://github.com/stoffern/react-native-image-cropper

**Comment by superandrew213** [comment]
yes I'm playing around with both options testing which one performs the best

**Comment by stoffern** [comment]
@superandrew213 i know there is a bug with the `pan to move` in my cropper.
I have fixed it just not commited the update yet :smiley:

**Comment by stoffern** [comment]
@jeanpan not to be a copycat or anything but after a view of the code i found out that selecting items would trigger a complete rerender of the SGListView.
With that in mind, and the batchSize added from your side and my request of having loading spinners for all images i started to a total rewrite of the component.

https://github.com/stoffern/react-native-media-picker
Please try it and see if you would like to merge it?
It has quite a performance boost as selecting items only trigger render of the item itself.

**Comment by jeanpan** [comment]
@stoffern thanks for your suggestion. I will prepare a commit for fixing the problem of rerender SGListView.

**Comment by shireen** [comment]
How do I use the newly added loading indicator?

**Comment by BonnieMilian** [comment]
Dinamic ImageItem ReSize when the Layout Orientation change.
